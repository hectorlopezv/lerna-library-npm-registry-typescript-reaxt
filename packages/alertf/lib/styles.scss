.alert--inlineBold {
  font: normal normal bold 14px/16px Roboto;
  letter-spacing: 0px;
  color: #3686af;
}

.alert-custom {
  display: grid !important;
  width: 453.5px;
  padding: 3px !important;
  margin: 15px auto;

  background-color: #d9edf7;
  border-color: #bce8f1;
  color: #3686af !important;

  place-items: center center;

  grid-gap: 5px;
  grid-template-columns: 57px 286px 17px 32px;

  grid-template-rows: 1fr;
  justify-content: center;
  align-content: center;
  @media screen and (max-width: 767px) {
    width: 252px;
    margin: 10px auto 14px -6px;
    grid-template-columns: 27px 65px 94px 36px;
    padding: 2px 4px 12px 4px !important;
  }

  & > i {
    font-size: 50px;
    padding: 8px;

    @media screen and(max-width: 767px) {
      grid-column: 1 / span 3;
      padding: 8px 8px 8px 22px;
    }

    &::before {
      padding: 0px 10px 0px 2px !important;
    }
  }

  .alert__message {
    line-height: 1.5em;
    font: normal normal normal 14px/16px Roboto;
    letter-spacing: 0px;
    color: #3686af;
    @media screen and (max-width: 767px) {
      grid-row: 2 / span 1;
      grid-column: 2 / span 3;
      line-height: 14px;
    }
    & > span {
      display: inline-block;
    }
  }

  .close {
    text-shadow: none;
    filter: alpha(opacity=100);
    opacity: 1;
    font-size: 34px !important;
    padding: 0px 10px;
    grid-column: 4 / span 1;

    &.alert-success {
      color: #43894e;
    }

    &.alert-info {
      color: #3686af;
      background-color: transparent;
    }

    &.alert-warning {
      color: #8a6d3b;
    }

    &.alert-danger {
      color: #bb4945;
    }
  }
}
