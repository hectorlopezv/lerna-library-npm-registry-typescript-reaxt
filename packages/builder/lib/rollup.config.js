import { nodeResolve } from "@rollup/plugin-node-resolve";
import commonjs from "@rollup/plugin-commonjs";
import image from "@rollup/plugin-image";
import json from "@rollup/plugin-json";
import { terser } from "rollup-plugin-terser";
import peerDepsExternal from "rollup-plugin-peer-deps-external";

import postcss from "rollup-plugin-postcss";
import autoprefixer from "autoprefixer";
import cssnano from "cssnano";
import ts from "rollup-plugin-ts";
import sourcemaps from "rollup-plugin-sourcemaps";
import path from "path";
import process from "process";

const currentWorkingPath = process.cwd();

const { filename, devDependencies } = require(path.join(
  currentWorkingPath,
  "package.json"
));
console.log("filename", filename);
console.log("devDependencies", devDependencies);
const inputPath = path.join(
  currentWorkingPath,
  `lib/${filename[0]}.${filename[1]}`
);
const globals = {
  ...devDependencies,
};
const configPostCss = postcss({
  plugins: [
    cssnano({
      preset: "default",
    }),
    autoprefixer(),
  ],
  inject: {
    insertAt: "top",
  },
  extract: false,
});

export default {
  input: inputPath,
  output: [
    {
      file: `dist/${filename[0]}.cjs.js`,
      format: "cjs",
      sourcemap: true,
      plugins: [terser()],
    },
    {
      file: `dist/${filename[0]}.esm.js`,
      format: "esm",
      sourcemap: true,
      plugins: [terser()],
    },
  ],
  plugins: [
    peerDepsExternal(),
    json(),
    configPostCss,
    image(),
    commonjs({
      exclude: "node_modules",
      ignoreGlobal: true,
    }),
    ts({
      tsconfig: {
        allowJs: true,
        allowSyntheticDefaultImports: true,
        declaration: true,
        declarationDir: "dist",
        esModuleInterop: true,
        outDir: "./dist",
        jsx: "react",
        lib: ["es6", "dom"],
        module: "esnext",
        moduleResolution: "node",
        noEmit: true,
        resolveJsonModule: true,
        rootDir: "./lib",
        sourceMap: true,
        strict: true,
        target: "es6",
      },
      include: ["lib/**/*"],
      exclude: ["**/node_modules", "**/rollup.config.js"],
    }),
    nodeResolve(),
    sourcemaps(),
  ],
  external: Object.keys(globals),
};
