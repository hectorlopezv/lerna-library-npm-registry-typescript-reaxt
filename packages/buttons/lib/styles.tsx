import styled from 'styled-components'

export const Container = styled.div`
  border-top: 1px dashed #ccc;
  padding-top: 2em;
  padding-bottom: 2em;
  display: flex;
  align-items: baseline;
  justify-content: center;
  width: 100%;
  margin-top: 25px;
  @media screen and (max-width: 767px) {
    flex-flow: column-reverse nowrap;
    align-items: center;
    padding-bottom: 1.5em;
    padding-top: 2.3em;
    margin-top: 50px;
  }
`

export const Limpiar = styled.input`
  &&& {
    font: normal normal 300 14px/16px Roboto;
    text-align: center;
    background-color: transparent;
    color: #ff6600ba;
    text-transform: uppercase;
    padding: 5px 10px;
    text-align: center;
    border: none;
    @media screen and (max-width: 767px) {
      margin-left: 0px;
    }

    &:disabled {
      background: transparent;
      border: none;
      color: #7779;
      cursor: not-allowed;
    }
  }
`

export const Generate = styled.input`
  &&& {
    transition: background-color 0.2s ease-out;
    padding: 10px 14px 11px 14px;
    font: normal normal bold 14px/16px Roboto;
    letter-spacing: 0px;
    border-radius: 5px;
    text-align: center;
    background-color: #f60;
    border-color: transparent;
    color: #fff;

    &:disabled {
      opacity: 1;
      color: #999;
      background-color: #e5e5e5;
      border: solid 1px #ccc;
      cursor: not-allowed;
      &:hover {
        background-color: #e5e5e5;
      }
    }
    &:hover {
      background: #777;
    }
    @media screen and (min-width: 359px) {
      margin-bottom: 12px;
    }
  }
`
