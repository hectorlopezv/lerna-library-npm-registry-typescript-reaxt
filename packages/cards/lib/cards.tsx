import React, {ReactNode} from 'react'

import {
  CardsContainer,
  TitleContainer,
  Title,
  ChildrenContainer,
} from './styles'

export interface CardsProps {
  children: ReactNode
  title: string
  classExtra?: string
}

export const Cards: React.FC<CardsProps> = ({
  classExtra = '',

  title = '',
  children,
}) => {
  return (
    <CardsContainer classExtra={classExtra} data-testid="cards container">
      {title && (
        <TitleContainer>
          <Title>{title}</Title>
        </TitleContainer>
      )}

      {children && <ChildrenContainer>{children}</ChildrenContainer>}
    </CardsContainer>
  )
}
