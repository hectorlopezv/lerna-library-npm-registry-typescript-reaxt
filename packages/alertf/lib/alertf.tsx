import React from 'react'

import './styles.scss'

export interface IAlertFProps {
  type: string
  mensaje: string
  handleFlyerClose?: (...args: unknown[]) => unknown
  className?: string
}

export class Alertf extends React.Component<IAlertFProps> {
  type = ''
  icon = ''
  setTypeClass = (input: string): void => {
    switch (input) {
      case 'info':
        this.type = 'alert-info'
        this.icon = 'fa-info-circle'
        break
      case 'success':
        this.type = 'alert-success'
        this.icon = 'fa-check-circle'
        break
      case 'warning':
        this.type = 'alert-warning'
        this.icon = 'fa-exclamation-circle'
        break
      case 'danger':
        this.type = 'alert-danger'
        this.icon = 'fa-exclamation-triangle'
        break
      default:
        this.type = 'alert-info'
        this.icon = 'fa-info-circle'
    }
  }

  componentWillMount() {
    this.setTypeClass(this.props.type)
  }
  public static defaultProps = {
    type: 'alert-info',
    className: '',
  }
  render() {
    return (
      <div
        className={`alert alert-custom alert-dismissible ${this.props.className}`}
        role="alert"
      >
        <i className={`fa ${this.icon}`}></i>

        <div className="alert__message">{this.props.mensaje}</div>

        <a
          className={`close ${this.type}`}
          data-bs-dismiss="alert"
          aria-label="Close"
          onClick={this.props.handleFlyerClose}
        >
          <i className="fa fa-times-circle"></i>
        </a>
      </div>
    )
  }
}
