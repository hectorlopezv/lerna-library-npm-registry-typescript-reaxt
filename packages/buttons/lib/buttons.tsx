import {Container, Limpiar, Generate} from './styles'
import React from 'react'

export interface Buttons_SCProps {
  className?: string
  show: [boolean, boolean]
  disableMessage?: boolean
  cleanMessage?: string
  handleCancel?: (...args: any[]) => any
  generateMessage?: string
  disableGenerate?: boolean
  querySickLeaves?: (...args: any[]) => any
}

export const Buttons_SC: React.FC<Buttons_SCProps> = ({
  className,
  show,
  disableMessage,
  cleanMessage,
  handleCancel,
  generateMessage,
  disableGenerate,
  querySickLeaves,
}) => {
  return (
    <Container className={className}>
      {show[0] === true && (
        <Limpiar
          type="button"
          disabled={disableMessage}
          value={cleanMessage}
          onClick={handleCancel}
        />
      )}

      {show[1] === true && (
        <Generate
          type="button"
          value={generateMessage}
          disabled={disableGenerate}
          onClick={querySickLeaves}
        />
      )}
    </Container>
  )
}
