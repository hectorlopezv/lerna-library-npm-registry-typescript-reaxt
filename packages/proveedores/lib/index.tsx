import {Smilef} from '@libProv/smilef'
import {Alertf} from '@libProv/alertf'
import {Selector} from '@libProv/selector'
import {FilterSelect} from '@libProv/filter'
import {Paginator} from '@libProv/paginator'
import {Loader} from '@libProv/loader'
import {ModalMessage} from '@libProv/message'
import {SliderTable} from '@libProv/dots'
import {Section} from '@libProv/section'
import {Phase1} from '@libProv/phase1'
import {Phase2} from '@libProv/phase2'
export {
  Smilef,
  Alertf,
  Selector,
  FilterSelect,
  Paginator,
  Loader,
  SliderTable,
  ModalMessage,
  Section,
  Phase1,
  Phase2,
}
