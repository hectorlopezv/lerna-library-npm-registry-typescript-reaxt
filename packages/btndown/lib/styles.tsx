import styled from 'styled-components'
import FaFileArchiveIcon from './FaFileArchiveIcon'
import {Buttons} from '@artefactos/buttons'
interface propsZipper {
  tabIndex: number
}
export const Container = styled.div`
  display: flex;
  align-items: baseline;
  margin-top: 20px;

  @media screen and (max-width: 767px) {
    flex-flow: column;
    align-items: center;
    margin-top: -6px;
    margin-bottom: 13px;
  }
`

export const Inner = styled.div`
  display: flex;
  align-items: center;
  //margin-left: 34px;
  margin-bottom: 10.6px;
  align-self: flex-end;
  margin-bottom: 17.5px;
  @media screen and (max-width: 767px) {
    flex-flow: column;
    margin: 0;
    margin-top: 22px;
    align-self: center;
    margin-bottom: 15px;
  }
`

export const IconFile = styled(FaFileArchiveIcon)`
  //margin-left: 29px;
  font-size: 24px;
  @media screen and (max-width: 767px) {
    margin-left: 0px;
    margin-top: 0px;
  }
`
export const Label = styled.span.attrs<propsZipper>(prop => ({
  tabindex: prop.tabIndex,
}))`
  margin-left: 7px;
  font: normal normal 300 14px/16px Roboto;

  @media screen and (max-width: 767px) {
    margin-left: 0px;
    margin-top: 5px;
  }
`
export const Button = styled(Buttons)`
  &&& {
    margin-left: auto;
    border: none;
    width: unset;
    /* margin-right: 43px; */
    padding: 0;
    margin-right: 13px;
    @media screen and (max-width: 767px) {
      margin-left: 0px;
      margin: 0;
      margin-right: 5px;
    }
  }
`

export const ContainerEl = styled.div`
  &&& {
    margin-left: 56px;
    margin-top: 30px;
    margin-right: 19px;
    @media screen and (max-width: 768px) {
      margin-top: 56px;
      margin-left: 0;
      margin-right: 0px;
      text-align: center;
      padding: 0 16px;
    }
  }
`

export const TitleZipper = styled.h3<propsZipper>`
  &&& {
    font: normal normal normal 17.6px/21px Roboto;
    letter-spacing: 0px;
    color: #999999;
  }
`
