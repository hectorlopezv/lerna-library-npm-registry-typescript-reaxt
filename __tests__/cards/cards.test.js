import React from 'react'
import {render, screen} from '@testing-library/react'

import {Cards} from '@libprov/cards/lib/cards'

describe('testing suite', () => {
  it('testing unit Cards', () => {
    render(
      <Cards title="hello im card title" classExtra="im class">
        <div data-testid="hello">hello im Here</div>
        <div data-testid="hello">hello im Here2</div>
      </Cards>,
    )

    expect(
      screen.getByRole('heading', {
        name: /hello im card title/i,
      }),
    ).toHaveTextContent(/hello im card title/i)

    expect(screen.queryAllByTestId(/hello/i)).toHaveLength(2)
  })
})
