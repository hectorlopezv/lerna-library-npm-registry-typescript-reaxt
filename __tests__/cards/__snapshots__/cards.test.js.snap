// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing suite testing unit Cards 1`] = `
<h2
  class="sc-htpNat gKEsUm"
>
  hello im card title
</h2>
`;
